src/App.tsx

```diff
@@ -15,48 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getSortedGoods(goodsArr: string[], sortField: SortType): string[] {
+  if (sortField !== SortType.Default) {
+    goodsArr.sort((a, b): number => {
+      switch (sortField) {
+        case SortType.Alphabet: {
+          return a.localeCompare(b);
+        }
+
+        case SortType.Length: {
+          return a.length - b.length;
+        }
+
+        default: {
+          return 0;
+        }
+      }
+    });
+  }
+
+  return goodsArr;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState(SortType.Default);
+  const isHideResetBtn = sortField || isReversed;
+
+  const resetField = () => {
+    setIsReversed(false);
+    setSortField(SortType.Default);
+  };
+
+  const goods = getSortedGoods([...goodsFromServer], sortField);
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SortType.Alphabet)}
+          className={cn(
+            'button is-info', { 'is-light': sortField !== SortType.Alphabet },
+          )}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          onClick={() => setSortField(SortType.Length)}
+          className={cn(
+            'button is-success', { 'is-light': sortField !== SortType.Length },
+          )}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          onClick={() => setIsReversed(!isReversed)}
+          className={cn('button is-warning',
+            { 'is-light': !isReversed })}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {isHideResetBtn && (
+          <button
+            type="button"
+            onClick={() => resetField()}
```

Comment:
Such places can be recorded more easily:
```suggestion
            onClick={resetField}
```